src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -533,43 +533,67 @@ public static int findNumberOfRepetitions(byte[] repeatUnit, byte[] testString,
         if (testString.length == 0){
             return 0;
         }
-        final int repeatLength = repeatUnit.length;
-        final int testLength = testString.length;
-        final int lengthDifference = testLength - repeatLength;
+        return findNumberOfRepetitions(repeatUnit, 0, repeatUnit.length, testString, 0, testString.length, leadingRepeats);
+    }
+
+    /**
+     * Finds number of repetitions a string consists of.
+     * Same as {@link #findNumberOfRepetitions} but operates on subarrays of a bigger array to save on copying.
+     * For example, for string ATAT and repeat unit AT, number of repetitions = 2
+     * @param repeatUnitFull             Non-empty substring represented by byte array
+     * @param offsetInRepeatUnitFull     the offset in repeatUnitFull from which to read the repeat unit
+     * @param repeatUnitLength           length of the repeat unit
+     * @param testStringFull             string to test (represented by byte array), may be empty
+     * @param offsetInTestStringFull     the offset in offsetInRepeatUnitFull from which to read the test string
+     * @param testStringLength           length of the test string
+     * @param leadingRepeats         Look for leading (at the beginning of string) or trailing (at end of string) repetitions
+     * For example:
+     *    GATAT has 2 trailing repeats of A but 0 leading repeats of AT
+     *    ATATG has 0 leading repeats of A but 2 forward repeats of AT
```

Comment:
Are there 2 concepts here or 3?  Leading repeats, trailing repeats, forward repeats?  I'm assuming forward and leading are the same?  Lets choose one terminology.